File: auction_service/src/main/java/com/example/auction/model/dao/UserEntity.java
package com.example.auction.model.dao;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "users")
@Data
public class UserEntity {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long Id;

    private String emailAddress;

    private String password;

    private String name;

    private String lastName;

    @JoinColumn
    @OneToOne(cascade = {CascadeType.ALL})
    private AddressEntity addressEntity;

    private LocalDate dateOfCreatingUser;

    private String status;

    private String type;

    @OneToOne
    private AuthorityEntity authorityEntity;

}

File: auction_service frontend/auction/src/app/UserEntity.ts
import { AddressEntity } from './AddressEntity';

export class UserEntity {

  emailAddress: string;
  password: String;
  name: String;
  lastName: String;
  address: AddressEntity;

  constructor(emailAddress: string, password: String, name: String, lastName: String, address: AddressEntity) {
    this.emailAddress = emailAddress;
    this.password = password;
    this.name = name;
    this.lastName = lastName;
    this.address = address;
  }

}